test/codegen/fromcto/typescript/typescriptvisitor.js

```diff
@@ -629,6 +641,100 @@ describe('TypescriptVisitor', function () {
+
+            sinon.stub(ModelUtil, 'isScalar').callsFake(() => {
+                return false;
+            });
+
+            getKeyType.returns('String');
+            getValueType.returns('String');
+            mockMapDeclaration.getName.returns('Map1');
+            mockMapDeclaration.isMapDeclaration.returns(true);
+            mockMapDeclaration.getKey.returns({ getType: getKeyType });
+            mockMapDeclaration.getValue.returns({ getType: getValueType });
+
+            typescriptVisitor.visitMapDeclaration(mockMapDeclaration, param);
+
+            param.fileWriter.writeLine.withArgs(1, 'export type Map1 = Map<string, string>;').calledOnce.should.be.ok;
+        });
+
+        it('should write a line with the name, key and value of the map <String, DateTime>', () => {
+            let param = {
+                fileWriter: mockFileWriter
+            };
+
+            let mockMapDeclaration = sinon.createStubInstance(MapDeclaration);
+
+            const getKeyType = sinon.stub();
+            const getValueType = sinon.stub();
+
+            getKeyType.returns('String');
+            getValueType.returns('DateTime');
+            mockMapDeclaration.getName.returns('Map1');
+            mockMapDeclaration.isMapDeclaration.returns(true);
+            mockMapDeclaration.getKey.returns({ getType: getKeyType });
+            mockMapDeclaration.getValue.returns({ getType: getValueType });
+
+            typescriptVisitor.visitMapDeclaration(mockMapDeclaration, param);
+
+            param.fileWriter.writeLine.withArgs(1, 'export type Map1 = Map<string, Date>;').calledOnce.should.be.ok;
+        });
+
+        it('should write a line with the name, key and value of the map <String, Address>', () => {
+            let param = {
+                fileWriter: mockFileWriter
+            };
+
+            let mockMapDeclaration = sinon.createStubInstance(MapDeclaration);
+
+            const getKeyType = sinon.stub();
+            const getValueType = sinon.stub();
+
+            getKeyType.returns('String');
+            getValueType.returns('Address');
+            mockMapDeclaration.getName.returns('Map1');
+            mockMapDeclaration.isMapDeclaration.returns(true);
+            mockMapDeclaration.getKey.returns({ getType: getKeyType });
+            mockMapDeclaration.getValue.returns({ getType: getValueType });
+
+            typescriptVisitor.visitMapDeclaration(mockMapDeclaration, param);
+
+            param.fileWriter.writeLine.withArgs(1, 'export type Map1 = Map<string, IAddress>;').calledOnce.should.be.ok;
+        });
+
+        it('should write a line with the name, key and value of the map <DateTime, Address>', () => {
+            let param = {
+                fileWriter: mockFileWriter
+            };
+
+            let mockMapDeclaration = sinon.createStubInstance(MapDeclaration);
+
+            const getKeyType = sinon.stub();
+            const getValueType = sinon.stub();
+
+            getKeyType.returns('String');
```

Comment:
Should this return `DateTime`?

Reply:
@ekarademir see **Note** in the preview description.